src/nvim/tui/input.c

```diff
@@ -106,17 +106,15 @@ static void tinput_wait_enqueue(void **argv)
   RBUFFER_UNTIL_EMPTY(input->key_buffer, buf, len) {
     const String keys = { .data = buf, .size = len };
     if (input->paste) {
-      Error err = ERROR_INIT;
-      // Paste phase: "continue" (unless handler canceled).
-      input->paste = !nvim_paste(keys, true, input->paste, &err)
-        ? 0 : (1 == input->paste ? 2 : input->paste);
+      String copy = copy_string(keys);
+      multiqueue_put(main_loop.events, tinput_paste_event, 3,
+                     copy.data, copy.size, (intptr_t)input->paste);
+      // started paste, further chunks should use 2
```

Comment:
other phase transitions have "Paste phase:" comment prefix.

```suggestion
      // Paste phase: "continue"
```